flow-nodes/api-builder-plugin-format-date/src/actions.js

```diff
@@ -3,23 +3,33 @@ const moment = require('moment');
 /**
  * Action method.
  *
- * @param {object} req - The flow request context passed in at runtime.  The
- *	 parameters are resolved as `req.params` and the available authorization
- * credentials are passed in as `req.authorizations`.
- * @param {object} outputs - A set of output callbacks.  Use it to signal an
- *	 event and pass the output result back to the runtime.  Only use an
- *	 output callback once and only after all asyncronous tasks complete.
- *
- * @return {undefined}
+ * @param {object} params - A map of all the parameters passed from the flow.
+ * @param {object} options - The additional options provided from the flow
+ *	 engine.
+ * @param {object} options.pluginConfig - The service configuration for this
+ *	 plugin from API Builder config.pluginConfig['api-builder-plugin-pluginName']
+ * @param {object} options.logger - The API Builder logger which can be used
+ *	 to log messages to the console. When run in unit-tests, the messages are
+ *	 not logged.  If you wish to test logging, you will need to create a
+ *	 mocked logger (e.g. using `simple-mock`) and override in
+ *	 `MockRuntime.loadPlugin`.  For more information about the logger, see:
+ *	 https://docs.axway.com/bundle/API_Builder_4x_allOS_en/page/logging.html
+ * @param {*} [options.pluginContext] - The data provided by passing the
+ *	 context to `sdk.load(file, actions, { pluginContext })` in `getPlugin`
+ *	 in `index.js`.
+ * @return {*} The response value (resolves to "next" output, or if the method
+ *	 does not define "next", the first defined output).
  */
-function formatDate(req, outputs) {
-	const { date, format } = req.params;
+function formatDate(params, { logger }) {
+	const { date, format } = params;
 
 	if (!date) {
-		return outputs.error(null, new Error('Missing required parameter: date'));
+		logger.error('The date parameter is missing.');
+		throw new Error('Missing required parameter: date');
 	}
 	if (!format) {
-		return outputs.error(null, new Error('Missing required parameter: format'));
+		logger.error('The format parameter is missing.');
```

Comment:
additional logger was there by default while creating a new plugin, so it was added here. So are we using it to show the extra information which is not showing in the regular flow?

Reply:
The regular flow logs should already show the data that was input and output from the flow so it will be logged twice (just slightly differently and at different levels), by logging it here. If you want to see the flow logs, set the log level to debug or higher in API Builder